main/display.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_timer.h"
+#include "esp_lcd_panel_io.h"
+#include "esp_lcd_panel_ops.h"
+#include "esp_err.h"
+#include "esp_log.h"
+#include "esp_check.h"
+#include "lvgl.h"
+#include "lvgl__lvgl/src/themes/lv_theme_private.h"
+#include "esp_lvgl_port.h"
+#include "global_state.h"
+#include "nvs_config.h"
+#include "i2c_bitaxe.h"
+#include "driver/i2c_master.h"
+#include "driver/i2c_types.h"
+#include "esp_lcd_panel_ssd1306.h"
+
+#define SSD1306_I2C_ADDRESS    0x3C
+
+#define LCD_H_RES              128
```

Comment:
as you suggested in dc already, we might need to figure out a good way for having different display sizes

Reply:
See #572.